ambry-frontend/src/main/java/com/github/ambry/frontend/FrontendRestRequestService.java

```diff
@@ -293,8 +296,13 @@ public void handleGet(final RestRequest restRequest, final RestResponseChannel r
             (result, exception) -> submitResponse(restRequest, restResponseChannel, result, exception));
       } else if (requestPath.matchesOperation(Operations.NAMED_BLOB)
           && NamedBlobPath.parse(requestPath, restRequest.getArgs()).getBlobName() == null) {
-        namedBlobListHandler.handle(restRequest, restResponseChannel,
-            (result, exception) -> submitResponse(restRequest, restResponseChannel, result, exception));
+        if (isS3Request(restRequest)) {
```

Comment:
We are converting the path in the RequestPath so here we have to rely on the isS3Request method. What if this is an upload or download request? We would have to use directly NamedBlob handlers directly?

Reply:
sure